apostrophecms/apostrophe
# Pro 3112 extend attachment filegroups

## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*

The `attachment` module currently has an option, `fileGroups` that will completely replace the default `fileGroups` of `office` and `images` with the user-provided groups. This is inconvenient if the user only wants to add a single new extension to the existing groups or add a new group while retaining the rest of the default groups. This PR modifies the `attachment` module to allow the addition of an extension(s) to the existing groups or a new group without discarding the default groups. It also contains two new tests for this behavior. Closes PRO-3112.

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly

1. Create a `modules/@apostrophecms/attachment/index.js` file.
2. Add the following code:
```
module.exports = {
  options: {
    addFileGroups: [
      {
        name: 'office',
        extensions: [
          'odt'
        ]
      },
      {
        name: 'othermedia',
        label: 'media',
        extensions: [
          'mp4'
        ],
        extensionMaps: {}
      }
    ]
  }
};

```
3. Bring up a site and check that you can still upload an existing `office` group file like a 'pdf'.
4. Check you can upload an 'odt' file.
5. Check you can upload an 'mp4' file.
6. Run the tests and check that the new tests pass correctly.

## What kind of change does this PR introduce?
*(Check at least one)*


## Make sure the PR fulfills these requirements:


If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**
The related documentation will be updated once this PR is approved. The documentation will not be merged until the release data of this improvement.
